faq/management/commands/zendesk_student_faq.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+
+import datetime
+import dateutil.parser
+import json
+import requests
+import pymongo
+import pytz
+
+from django.conf import settings
+from django.utils import timezone
+
+from optparse import make_option
+
+from django.core.management.base import BaseCommand, CommandError
+
+from ...utils import connect_to_mongo, COLLECTION
+
+# zendesk API
+BASE = 'https://fun1.zendesk.com'
+USERNAME = 'laurent.at.fun@gmail.com'
+TOKEN = '9flZe3sgFn4JIu4ha1K7WTb5V8W5ypniBNZenAe9'
+
+
+class Command(BaseCommand):
+    help = """Retrieve from Zendesk API the student FAQ and store it to mongo so we can display it on website.
+    """
+    option_list = BaseCommand.option_list + (
+        make_option('-p', '--print',
+                    action='store_true',
+                    dest='print',
+                    default=False),
+        make_option('-d', '--dry-run',
+                    action='store_true',
+                    dest='dry_run',
+                    default=False),)
+
+
+    def _query_zendesk(self, url):
+        response = requests.get(BASE + url, auth=(USERNAME + "/token", TOKEN))
+        data = json.loads(response.content)
+        return data
+
+    def handle(self, *args, **options):
+
+        category_count = 0
+        section_count = 0
+        article_count = 0
+        faq = {}
+        faq['articles'] = []
+        faq['categories'] = []
+        faq['last_update'] = timezone.now()  # This is the last time we successuly fetched Zendesk API
+        categories = self._query_zendesk('/api/v2/help_center/categories.json')
+
+        for category in categories['categories']:
+            if options['print']:
+                self.stdout.write("%s\n" % category['name'])
+            faq['categories'].append({
+                    'name': category['name'],
+                    'id': category['id'],
+                    'position': category['position'],
+                    'sections': []})
+            category_count +=1
+            sections = self._query_zendesk('/api/v2/help_center/categories/{id}/sections.json'.format(
+                    id=category['id']))
+            for section in sections['sections']:
+                if options['print']:
+                    self.stdout.write("%s\n" % section['name'])
+                faq['categories'][-1]['sections'].append({
+                    'id': section['id'],
+                    'name': section['name'],
+                    'position': section['position'],
+                    'articles': [],
+                    })
+                section_count +=1
+                articles = self._query_zendesk('/api/v2/help_center/sections/{id}/articles.json'.format(
+                        id=section['id']))
+
+                last_document_update = datetime.datetime(1970, 1, 1).replace(
+                        tzinfo=pytz.timezone('UTC'))  # initialize with a date in the far past
+
+                for article in articles['articles']:
```

Comment:
Il y a trois boucles `for`. Pour plus de lisibilité, on pourrait écrire des fonctions du genre : 

    faq['categories'].append(faq_category(category_name))
    ...